src/main/java/name/webdizz/sonar/grammar/spellcheck/JavaSourceCodeWordFinder.java

```diff
@@ -3,16 +3,22 @@
 import com.swabunga.spell.event.AbstractWordFinder;
 import com.swabunga.spell.event.Word;
 import com.swabunga.spell.event.WordNotFoundException;
+import name.webdizz.sonar.grammar.PluginParameter;
+import org.sonar.api.BatchExtension;
+import org.sonar.api.config.Settings;
 
-public class JavaSourceCodeWordFinder extends AbstractWordFinder {
+public class JavaSourceCodeWordFinder extends AbstractWordFinder implements BatchExtension {
     private boolean inComment;
+    private int minimumWordLength;
+    private Settings settings;
 
     public JavaSourceCodeWordFinder() {
         super();
     }
 
-    public JavaSourceCodeWordFinder(final String searchText) {
+    public JavaSourceCodeWordFinder(final String searchText, final Settings settings) {

```

Comment:
looks like this method never invoked, how can we make sure that settings will be injected? As i understood SonarQube uses default constructor to create a class instance

let's deep dive a bit tomorrow